app.js

```diff
@@ -60,37 +37,72 @@ let fetch = (user, repo) => {
       sort: "updated",
       order: "desc",
       since: ghCommit.commit.author.date
-    });
-  }).then((issues) => {
-    const json = {};
+    })
+  }
+}
+
+const testLabel = (l) => { return regex.test(l.name) }
+
+const fetchCommits = (user, repo) => {
+  return github.repos.getCommits({
+    user,
+    repo,
+    sha: "production"
+  })
+}
+
+const collectIssues = (issues) => {
+  return new Promise((resolve, reject) => {
```

Comment:
If it is asynchronous then this code is working by luck of resolving with an empty object reference that asynchronously gets filled in time for it to be used (since you resolve in a synchronous context) - what error do you get when you make it synchronous?